### ISSUE TITLE
ASM-related crash when generating extra village decorations.

### ISSUE THREAD

**Issue by Dakedres** [svenhjol/Charm#249]
When attempting to create a world in 1.12 I recieved the following crash:

[Log](https://gist.github.com/Dakedres/be8db26501a545668810d144df909b6c)
[Modlist](https://gist.github.com/Dakedres/cbc3cae25b110e322df418863ae5d360)

Forge version: 14.23.5.2847
Charm version: 1.3.7

**Comment by svenhjol** [comment]
The log doesn't help me debug it, I don't see the crash or stacktrace in it

**Comment by Dakedres** [comment]
Sorry, that was due to VanillaFix being weird. I looked a bit further into the issue and it was actually caused by a broken config from another mod, I thought it was Charm because it was pretty high in the stack trace and I thought disabling charm fixed it, but as I did some more testing I realized the issue still persisted. Sorry about that, have a nice day!

**Comment by Dakedres** [comment]
Okay, apparently that was a false alarm, the issue does persist, it was just more apparent because that other mod was making villages spawn so much more frequently.

[Crash (includes modlist)](https://gist.github.com/Dakedres/0219f166d9a4848f2c39b10a5d146b73)

I was initially using RTG, but then I tested it in the default vanilla world type and found that the crash still occurred, I also found that disabling "VillageDecorations" in the Charm config prevents the issue. Villages where this crash occurs are pretty rare, so I could send the modpack and a seed with a bugged village near spawn if that would help.

**Comment by svenhjol** [comment]
I'm still not able to get anything from that log. It's not a crash log, it's just a general log of things that are happening. Normally at the end of the log you'd get an error message + a stacktrace before it crashes out. Without that, I don't have anything to start debugging with.

**Comment by Dakedres** [comment]
My bad, clipboard limit made it copy and paste incorrectly, I'll get the full log asap.

**Comment by Dakedres** [comment]
Here is [the complete log](https://paste.ee/p/exnH0), and the isolated [crash](https://gist.github.com/Dakedres/f650a09e870cec0fb4fb6cd86a521b75) sorry about all this.

**Comment by Dakedres** [comment]
the fuck is wrong with me why do i keep click the close and comment holy shit

**Comment by faeldray** [comment]
I'm experiencing the same crash using Charm 1.3.7.

https://pastebin.com/DfUbABX0

Downgrading to 1.3.6 seems to stop the issue though.

**Comment by faeldray** [comment]
Nope, nevermind, 1.3.6 also has this problem. :-/

https://pastebin.com/Tt1vYxEp

**Comment by faeldray** [comment]
I did some more testing and I'm fairly certain I've determined the issue. It occurs when the Rats mod is installed AND when the pet shop is disabled in the config. If the pet shop is enabled, everything works fine.

It used to be that when the shop was disabled, the pet shop villager still existed. But a change was made not too long ago so that disabling the shop also disables the villager, which I'm guessing is somehow the problem here (https://github.com/Alex-the-666/Rats/commit/784610a40d778515aeb12c291e19900a71cb1c6a).

For reference: https://pastebin.com/WFUdEUyj
